src/EFCore.Jet/Query/ExpressionTranslators/Internal/JetByteArrayMethodTranslator.cs

```diff
@@ -21,6 +21,9 @@ public class JetByteArrayMethodTranslator : IMethodCallTranslator
 {
     private readonly ISqlExpressionFactory _sqlExpressionFactory;
 
+    private MethodInfo ByteArrayLength = typeof(JetDbFunctionsExtensions).GetRuntimeMethod(
+        nameof(JetDbFunctionsExtensions.ByteArrayLength),
+        new[] { typeof(DbFunctions), typeof(byte[]) })!;
```

Comment:
I wrote this online on github without checking the syntax, so please verify it first.

Also, I believe the wrong length can be returned for data with an _even_ original length, not an _odd_ length.

(The odd length data will always be appended with a `0x00` byte, which we then substract from the length. But the even length data will not get a `0x00` byte appended; if it actually ends with a `0x00` byte, the length we return will by one byte short.)

```c#
/// <summary>
///     Returns the length of <paramref name="byteArray"/>, or the length of <paramref name="byteArray"/> <c>-1</c>. If the actual length of <paramref name="byteArray"/> is even and the last byte is
///     <c>0x00</c>, the length of <paramref name="byteArray"/> <c>-1</c> is returned. Otherwise, the actual length of <paramref name="byteArray"/> is returned.
/// </summary>
/// <remarks>
///     <para>
///     Jet SQL reads byte arrays into strings. As Jet uses Unicode, the internal length will always be a multiple of 2. If your data has an odd number of bytes, Jet internally adds a <c>0x00</c> byte to
///     the end of the array.
///     </para>
///     <para>
///     This method will test if the last byte of the array is <c>0x00</c>. If it is <c>0x00</c>, it is assumed that the last byte was added by Jet to fill the array to an even number of bytes and the internal
///     length <c>-1</c> is returned. In all other cases, the internal length is returned.
///     </para>
///     <para>
///     If the actual data length is odd, this method will always return the original length, independent of the value of the last byte of the original data.
///     If the actual data length is even and the original data does not end with a <c>0x00</c> byte, this method will return the original length.
///     If the actual data length is even and the original data does end with a <c>0x00</c> byte, this method will return the original length <c>-1</c>.
///     </para>
///     <para>
///     The only way to always get the correct array length is to either only use odd array lengths with this extension method, or to use client evalution instead of this extension method.
///     </para>
/// </remarks>
/// <param name="_">The DbFunctions instance.</param>
/// <param name="byteArray">The `byte[]` array.</param>
/// <returns>The length of <paramref name="byteArray"/>, or the length of <paramref name="byteArray"/> <c>-1</c>.</returns>
public static int ByteArrayLength(
    [CanBeNull] this DbFunctions _,
    [NotNull] byte[] byteArray)
    => throw new InvalidOperationException(CoreStrings.FunctionOnClient(nameof(ByteArrayLength)));
```

Reply:
Looks pretty good. Did make some small changes. 3rd paragraph added `<br/>` between to make sure it all started on a new line. 4th para changed to `If your data will never end in <c>0x00</c> you can use this extension method safely, otherwise it is highly recommended to only use client evaluation.`